src/mj_utils_merge_mujoco_models.cpp

```diff
@@ -87,6 +87,17 @@ static void merge_mujoco_option(const std::string & fileIn, const pugi::xml_node
   }
 }
 
+static void merge_mujoco_extension(const std::string & fileIn, const pugi::xml_node & in, pugi::xml_node & out)
+{
+  merge_mujoco_node("extension", fileIn, in, out);
+  const auto & plugin = in.child("plugin");
+  if(plugin)
```

Comment:
I think you want to change this to: `for(const auto & plugin : in.children("plugin"))`

This will iterate over all children of `in` that have the `plugin` tag

Reply:
Thanks, and how should I add all these `plugin`s to `out`?

Reply:
This is solved by https://github.com/rohanpsingh/mc_mujoco/pull/48/commits/36e11c56af984ca9197b379b793728e7708d2c8c